src/main/java/hudson/plugins/clover/CloverPublisher.java

```diff
@@ -175,14 +199,105 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         return true;
     }
 
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws
```

Comment:
You can call this method from the deprecated one and avoid to duplicate a lot of code.